tests/unit/test_charm.py

```diff
@@ -161,6 +184,36 @@ def test_layer_updated_without_any_endpoint_info(harness: Harness) -> None:
     assert harness.charm._login_ui_layer.to_dict() == expected_layer
 
 
+def test_layer_updated_with_tracing_endpoint_info(harness: Harness) -> None:
+    """Test Pebble Layer when relation data is in place."""
+    harness.set_leader(True)
+    harness.set_can_connect(CONTAINER_NAME, True)
+    harness.charm.on.login_ui_pebble_ready.emit(CONTAINER_NAME)
+    tracing_relation_id = setup_tracing_relation(harness)
+
+    host = harness.get_relation_data(tracing_relation_id, "tempo-k8s")["host"]
+    http = filter(
+        lambda x: x["protocol"] == "otlp_http",
+        harness.get_relation_data(tracing_relation_id, "tempo-k8s")["ingesters"],
```

Comment:
Nitpick:

There are multiple calls to the `harness.get_relation_data(tracing_relation_id, "tempo-k8s")`. We can assign the result to a local variable to avoid multiple calls.